snafu/log_generator_wrapper/trigger_log_generator.py

```diff
@@ -0,0 +1,274 @@
+        return int(query_results['statistics']['recordsMatched'])
+
+    def _check_es(self,start_time,end_time):
+        logger.info("Checking ElasticSearch for expected messages")
+        header_json = 'Content-Type: application/json'
+        header_auth = "Authorization: Bearer " + self.es_token
+        s_time = datetime.datetime.fromtimestamp(start_time - 60).strftime("%Y-%m-%dT%H:%M:%S")
+        e_time = datetime.datetime.fromtimestamp(end_time + 60).strftime("%Y-%m-%dT%H:%M:%S")
+        data = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "match": {
+                                "message": self.my_message
+                            }
+                        }
+                    ],
+                    "filter": [
+                        {
+                            "range": {
+                                "@timestamp": {
+                                    "gte": s_time,
+                                    "lte": e_time
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+
+        es_url = self.es_url + "/" + self.es_index + "/_count"
+
+        try:
+            response = subprocess.check_output(['curl','--insecure','--header',header_auth,
+                                                '--header',header_json,es_url,'-d',json.dumps(data),
+                                                '-s']).decode("utf-8")
+        except Exception as err:
+            logging.info("ElasticSearch query failed")
+            logging.info(err)
+            return 0
+        try:
+            return(json.loads(response)['count'])
+        except Exception as err:
+            logging.info("No valid json returned")
+            logging.info(err)
+            return 0
+
+    def emit_actions(self):
+        logger.info("Running log test with %d byte size for %d minutes at a rate of %d messages per second" %
+                    (self.size, self.duration, self.messages_per_second))
+        logger.info("Test UUID is %s on cluster %s" % (self.uuid, self.cluster_name))
+
+        logger.info("Waiting for all pods to be ready")
+        # Check redis for all running pods
+        if self.redisip is not None\
+            and self.redisport is not None\
+            and self.pod_name is not None\
+            and self.uuid is not None\
+            and self.pod_count != 1:
+            my_redis = redis.StrictRedis(
+                host=self.redisip,
+                port=self.redisport,
+                charset="utf-8",
+                decode_responses=True)
+            channel = "log-generator-" + self.uuid
+            self._wait_for_pods(my_redis,channel)
+
+        timestamp = time.strftime("%Y-%m-%dT%H:%M:%S")
+        start_time = time.time()
```

Comment:
Maybe it's better to use datetime.datetime.now() ?, I'm suggesting this because this way you can subtract dates natively (w/o converting them to epoch)

```python
start_time = datetime.datetime.now()
message_count = self._run_log_test()
end_time = datetime.datetime.now()
elapsed_time = (end_time - start_time).total_seconds()

Reply:
@rsevilla87 fair point. It functionally works out the same though in terms of the end result. I'll play around with it after the demo today since I don't want to break it prior :-D